sites/site-utilities/src/components/editor/index.tsx

```diff
@@ -0,0 +1,319 @@
+import React from "react";
+import ReactDOM from "react-dom";
+import { html_beautify } from "vscode-html-languageservice/lib/esm/beautify/beautify-html";
+import { mapDataDictionaryToMonacoEditorHTML } from "@microsoft/fast-tooling/dist/data-utilities/monaco";
+import {
+    AjvMapper,
+    CustomMessageIncomingOutgoing,
+    DataDictionary,
+    MessageSystem,
+    MessageSystemType,
+} from "@microsoft/fast-tooling";
+import { ViewerCustomAction } from "@microsoft/fast-tooling-react";
+// This is only used as a typescript reference, the actual monaco import must
+// be passed from the derived class or it will cause import issues
+import * as monaco from "monaco-editor/esm/vs/editor/editor.api";
+import { StandardLuminance } from "@microsoft/fast-components-styles-msft";
+import { classNames, Direction } from "@microsoft/fast-web-utilities";
+import FASTMessageSystemWorker from "@microsoft/fast-tooling/dist/message-system.min.js";
+import { MonacoAdapter } from "@microsoft/fast-tooling/dist/message-system-service/monaco-adapter.service";
+import {
+    MonacoAdapterAction,
+    MonacoAdapterActionCallbackConfig,
+} from "@microsoft/fast-tooling/dist/message-system-service/monaco-adapter.service-action";
+import { schemaDictionary } from "../../schemas";
+import { EditorState } from "./editor.props";
+
+export const previewBackgroundTransparency: string = "PREVIEW::TRANSPARENCY";
+export const previewDirection: string = "PREVIEW::DIRECTION";
+export const previewTheme: string = "PREVIEW::THEME";
+
+const fastMessageSystemWorker = new FASTMessageSystemWorker();
+
+abstract class Editor<P, S extends EditorState> extends React.Component<P, S> {
+    public editor: monaco.editor.IStandaloneCodeEditor;
+    public abstract editorContainerRef: React.RefObject<HTMLDivElement>;
+    public abstract viewerContainerRef: React.RefObject<HTMLDivElement>;
+    public viewerContentAreaPadding: number = 20;
+    public maxViewerHeight: number = 0;
+    public maxViewerWidth: number = 0;
+    public fastMessageSystem: MessageSystem;
+    public monacoValue: string[];
```

Comment:
Most of these fields seem pretty self-explanitory but I'm not sure what "monacoValue" is or what it should be. Could you add a comment to clarify?